include/fail_safes.hpp

```diff
@@ -0,0 +1,95 @@
+#include "config.hpp"
+#include "motors.hpp"
+
+namespace bzzz
+{
+    class FailSafes
+    {
+        private:
+        /**
+         * Radio timeout variables for connection loss check
+        */
+       unsigned long long m_lastRadioReceptionTime;
+       unsigned long m_radioConnectionTimeoutInMicroseconds;
+       bool m_HALT_SYSTEM = false;  // Bool to keep keep track if any fail-safe triggers a system halt.
+
+       /**
+        * Poiters to useful objects that are declared outside of this scope
+        * 
+        * Current list of objects pointer:
+        * 1. of class MotorDriver -> this is declared in main.cpp,
+        *                           used to control motors.
+       */
+       MotorDriver *motorDriver;
+
+       /**
+        * Check the radio connection and return true if it timed out 
+        * as follows:
+        * 
+        * if (micros - lastRadioreceptionTime > timeout) then the connection has been lost 
+        * and return false; otherwise return true
+        * 
+        * The function uses the system start time as reference and micros is the time since the 
+        * system was up.
+       */
+       bool radioConnectionCheck();
+
+       /**
+        * Shutdown the entire system
+        * 
+        * Currently only stalls the motors,
+        * might need to include a proper shutdown procedure in the future
+       */
+       void shutDown();
+
+       public:
+       /**
+        * Default constructor
+       */
+       FailSafes();
```

Comment:
I'm going to make this constructor private. One shouldn't be able to create a failsafe object with a null motor driver.